src/test/java/wolox/training/BookRepositoryTest.java

```diff
@@ -0,0 +1,113 @@
+package wolox.training;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.test.context.junit4.SpringRunner;
+import wolox.training.model.Book;
+import wolox.training.repositories.BookRepository;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = Replace.NONE)
+public class BookRepositoryTest {
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    @Autowired
+    private TestEntityManager entityManager;
+
+    private Book oneTestBook;
+
+    @Before
+    public void setUp() {
+        oneTestBook = new Book();
+        oneTestBook.setAuthor("Carlitos");
+        oneTestBook.setGenre("Terror");
+        oneTestBook.setImage("unaImagen");
+        oneTestBook.setIsbn("4578-8665");
+        oneTestBook.setPages(259);
+        oneTestBook.setPublisher("LaGuitarra");
+        oneTestBook.setSubtitle("Carlito's Way");
+        oneTestBook.setTitle("Las aventuras terrorificas de Carlitos");
+        oneTestBook.setYear("2005");
+        bookRepository.save(oneTestBook);
+    }
+
+    @Test
+    public void whenFindBookByAuthorAndExists_thenBookIsReturned() {
+        Book bookFromDatabase = bookRepository.findByAuthorIgnoreCase("Carlitos").orElseGet(null);
+        assertThat(bookFromDatabase).isEqualToComparingFieldByField(oneTestBook);
+    }
+
+    @Test
+    public void whenCreateBook_thenBookIsPersisted() {
+        Book persistedBook = bookRepository.findByAuthorIgnoreCase("Carlitos").orElse(null);
+        assertThat(persistedBook.getAuthor().equals(oneTestBook.getAuthor())).isTrue();
+        assertThat(persistedBook.getGenre().equals(oneTestBook.getGenre())).isTrue();
+        assertThat(persistedBook.getImage().equals(oneTestBook.getImage())).isTrue();
+        assertThat(persistedBook.getIsbn().equals(oneTestBook.getIsbn())).isTrue();
+        assertThat(persistedBook.getPages().equals(oneTestBook.getPages())).isTrue();
+        assertThat(persistedBook.getPublisher().equals(oneTestBook.getPublisher())).isTrue();
+        assertThat(persistedBook.getSubtitle().equals(oneTestBook.getSubtitle())).isTrue();
+        assertThat(persistedBook.getTitle().equals(oneTestBook.getTitle())).isTrue();
+        assertThat(persistedBook.getYear().equals(oneTestBook.getYear())).isTrue();
+    }
+
+    @Test(expected = NullPointerException.class)
```

Comment:
All these are throwing NPE because the checkNotNull of the model's setters are throwing it, and not because the save is getting executed with an error. The save never get's executed.
It's not wrong, just bear in mind that you're testing the models "business logic" with these tests and not the persistance layer.